### DISCUSSION TITLE
求教：umi4 构建产物有超多的文件，怎么合并产物到单个 js、css？

### DISCUSSION THREAD

**Discussion by 4x25** [umijs/umi#9377] [Q&A] [answered]
之前 umi3 的时候关闭 dynamicImport 就可以实现产物只有单个 js、css，现在 umi4 似乎删除了 dynamicImport 配置，请问现在该如何 合并产物到单个 js、css？

**Comment by 4x25** [comment, marked as answer]
原文：https://umijs.org/docs/introduce/faq#%E5%8F%AF%E4%BB%A5%E5%85%B3%E9%97%AD-dynamicimport-%E5%90%97

可以关闭 dynamicImport 吗？
可以，但不建议关闭。

1、安装依赖

  pnpm i babel-plugin-dynamic-import-node -D
2、配置里加上 extraBabelPlugins ，但只针对 production 开启

// .umirc.ts
export default {
  extraBabelPlugins: process.env.NODE_ENV === 'production' 
    ? ['babel-plugin-dynamic-import-node'] 
    : []
}

**Comment by hxmwr** [comment]
css 没有完全打包到一个文件里面，还有一些是以动态注入到 head 里面的 style 的形式存在

**Reply by 4x25** [reply]
> css 没有完全打包到一个文件里面，还有一些是以动态注入到 head 里面的 style 的形式存在

那是因为你用的是 antd 5
